PL2/PL2-shared/src/main/java/nl/tudelft/pl2016gr2/model/Node.java

```diff
@@ -55,12 +56,38 @@ public int hashCode() {
     return hash;
   }
 
-  public ArrayList<String> getGenomes() {
-    return genomes;
+  public Collection<String> getGenomes() {
+    return genomes.values();
   }
 
-  public void setGenomes(ArrayList<String> gs) {
-    this.genomes = gs;
+  /**
+   * Get the amount of genomes which travel from this node over an edge to the given node.
+   *
+   * @param to the node to which the edge goes.
+   * @return the amount of genomes which travel over the given edge.
+   */
+  @Override
+  public int getGenomesOverEdge(AbstractNode to) {
+    assert getOutlinks().contains(to.getId());
```

Comment:
`assert`s are not bad practice, they are similar to `exception`s, except (punny) that an `exception` may occur during run time. This function is an interaction with the data model. If we (the developers) call the model wrongly, we might trigger this assertion. However, this should **never possibly be triggered** in release code. Asserts in this projects are rather vital because they are not checked during the release, so they do not hinder performance. They still can trigger during development (when you compile with assert statements turned on), to notify of erreoneous behaviour.

In general: The caller is responsible for making sure that the provided `AbstractNode to` actually is an `outLink`. If you do not, you crash the application. Asserts just make it easier to see that you did :smile: